src/libraries/System.Text.Json/tests/Utf8JsonReaderTests.MultiSegment.cs

```diff
@@ -899,6 +899,82 @@ public static void TestSingleStringsMultiSegment()
             }
         }
 
+        [Fact]
+        public static void TestMultiSegmentStringConversionToDateTime()
+        {
+            string jsonString = "\"1997-07-16\"";
+            string expectedString = "1997-07-16";
+            int expectedTokenLength = 10;
+            DateTime expectedDateTime = DateTime.Parse(expectedString);
+
+            byte[] utf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            ReadOnlySequence<byte> sequence = JsonTestHelper.CreateSegments(utf8);
+
+            for (int j = 0; j < utf8.Length; j++)
+            {
+                var utf8JsonReader = new Utf8JsonReader(sequence.Slice(0, j), isFinalBlock: false, default);
+                ReadDateTimeHelper(ref utf8JsonReader, expectedDateTime, expectedTokenLength);
+
+                Assert.Equal(0, utf8JsonReader.TokenStartIndex);
+
+                long consumed = utf8JsonReader.BytesConsumed;
+                utf8JsonReader = new Utf8JsonReader(sequence.Slice(consumed), isFinalBlock: true, utf8JsonReader.CurrentState);
+                ReadDateTimeHelper(ref utf8JsonReader, expectedDateTime, expectedTokenLength);
+            }
+        }
+
+        private static void ReadDateTimeHelper(ref Utf8JsonReader jsonReader, DateTime expectedValue, long expectedTokenLength)
+        {
+            while (jsonReader.Read())
+            {
+                if (jsonReader.TokenType == JsonTokenType.String)
+                {
+                    long tokenLength = jsonReader.HasValueSequence ? jsonReader.ValueSequence.Length : jsonReader.ValueSpan.Length;
+                    Assert.Equal(expectedTokenLength, tokenLength);
+                    Assert.Equal(expectedValue, jsonReader.GetDateTime());
```

Comment:
nit: We missed the asserts for `TryGetDateTime` and `TryGetDateTimeOffset` APIs.